src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/AddKeepAlivesStep.cs

```diff
@@ -0,0 +1,103 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using Mono.Cecil;
+
+using Java.Interop.Tools.Cecil;
+
+using Mono.Linker;
+using Mono.Linker.Steps;
+
+using Mono.Cecil.Cil;
+
+namespace MonoDroid.Tuner
+{
+	public class AddKeepAlivesStep : BaseStep
+	{
+		readonly TypeDefinitionCache cache;
+
+		public AddKeepAlivesStep (TypeDefinitionCache cache)
+		{
+			this.cache = cache;
+		}
+
+		protected override void ProcessAssembly (AssemblyDefinition assembly)
+		{
+			if (AddKeepAlives (assembly)) {
+				AssemblyAction action = Annotations.HasAction (assembly) ? Annotations.GetAction (assembly) : AssemblyAction.Skip;
+				if (action == AssemblyAction.Skip || action == AssemblyAction.Copy || action == AssemblyAction.Delete)
+					Annotations.SetAction (assembly, AssemblyAction.Save);
+			}
+		}
+
+		internal bool AddKeepAlives (AssemblyDefinition assembly)
+		{
+			if (assembly.Name.Name != "Mono.Android" && !assembly.MainModule.HasTypeReference ("Java.Lang.Object"))
```

Comment:
@jonpryor, do we actually ever want to process `Mono.Android` assembly? I think when this step is run, the `Mono.Android` coming from the installation should be already "fixed".